### ISSUE TITLE
pipenv sync fails without Pipfile present

### ISSUE THREAD

**Issue by ulrikjohansson** [pypa/pipenv#2834]
##### Issue description

Running `pipenv sync` without a Pipfile present (the Pipfile.lock is there) fails with the message
```
Error: Pipfile.lock is missing! You need to run $ pipenv lock first.
```
##### Expected result
I would expect the Pipfile not to be needed at all for `sync` since it should just install the pinned versions from the Pipfile.lock

##### Actual result
pipenv won't install anything because it either thinks it needs a Pipfile as well as a Pipfile.lock for some reason. It also complains about the wrong file (Pipenv.lock).

##### Steps to replicate
* Create a new virtualenv, e.g with `pipenv install requests`
* Remove the Pipfile (keeping only the Pipfile.lock)
* Run `pipenv sync`

-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2018.7.1'`

Pipenv location: `'/usr/local/Cellar/pipenv/2018.7.1/libexec/lib/python3.7/site-packages/pipenv'`

Python location: `'/usr/local/Cellar/pipenv/2018.7.1/libexec/bin/python3.7'`

Other Python installations in `PATH`:

  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `2.7`: `/usr/local/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `3.7`: `/usr/local/bin/python3.7m`
  - `3.7`: `/usr/local/bin/python3.7`
  - `3.7`: `/usr/local/bin/python3.7`

  - `2.7.15`: `/usr/local/bin/python`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.15`: `/usr/local/bin/python`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.15`: `/usr/local/bin/python2`
  - `2.7.15`: `/usr/local/bin/python2`
  - `3.7.0`: `/usr/local/bin/python3`
  - `3.7.0`: `/usr/local/bin/python3`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.7.0',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '17.5.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 17.5.0: Fri Apr 13 19:32:32 PDT '
                     '2018; root:xnu-4570.51.2~1/RELEASE_X86_64',
 'python_full_version': '3.7.0',
 'python_version': '3.7',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `PATH`
  - `TERM_PROGRAM`
  - `SHELL`
  - `TERM`
  - `TMPDIR`
  - `PIPENV_VENV_IN_PROJECT`
  - `Apple_PubSub_Socket_Render`
  - `TERM_PROGRAM_VERSION`
  - `TERM_SESSION_ID`
  - `LC_ALL`
  - `USER`
  - `SSH_AUTH_SOCK`
  - `__CF_USER_TEXT_ENCODING`
  - `LSCOLORS`
  - `_`
  - `PWD`
  - `LANG`
  - `ITERM_PROFILE`
  - `XPC_FLAGS`
  - `XPC_SERVICE_NAME`
  - `COLORFGBG`
  - `HOME`
  - `SHLVL`
  - `ITERM_SESSION_ID`
  - `LOGNAME`
  - `LC_CTYPE`
  - `GOPATH`
  - `DISPLAY`
  - `COLORTERM`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_PYTHON_PATH`

Pipenv–specific environment variables:

 - `PIPENV_VENV_IN_PROJECT`: `1`

Debug–specific environment variables:

  - `PATH`: `/usr/local/Cellar/pipenv/2018.7.1/libexec/tools:/Users/ulrjoh2/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/opt/X11/bin:/Applications/Wireshark.app/Contents/MacOS:/usr/local/opt/go/libexec/bin:/Users/ulrjoh2/work/bin:/Users/ulrjoh2/.local/bin:/Users/ulrjoh2/.composer/vendor/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/opt/X11/bin:/Applications/Wireshark.app/Contents/MacOS:/usr/local/opt/go/libexec/bin:/Users/ulrjoh2/work/bin`
  - `SHELL`: `/bin/bash`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/tmp/test`

---------------------------

</details>

**Comment by ulrikjohansson** [comment]
From a quick look at the source, it seems that the reason for this behaviour is that pipenv figures out where `Pipfile.lock` is by first searching for `Pipfile`. When it can't find a Pipfile, it gives up and throws the error message about the missing lock file.

**Comment by uranusjr** [comment]
Honestly I am not sure `install --ignore-pipfile` should do after #1463 is fully implemented. There really should be a full-on review on various install commands and options. But from my understanding the original purpose of `--ignore-pipfile` is actually to ignore *the process of checking whether lock is up-to-date to Pipfile*, not Pipfile entirely (and no, it’s not named very intuitively), so it shouldn’t accept a missing Pipfile. The best behaviour is probably to error without one.

Review of the install command is far more involved than fixing the sync command. I would recommend creating a PEEP to fully discuss it. This thread would focus on fixing the sync command. The fix is not very difficult IMO, so I’ll slap a good first issue tag on it. Contribution are welcomed!

**Comment by ulrikjohansson** [comment]
I just found #1977 which seems to also mention this exact behaviour, see comment https://github.com/pypa/pipenv/issues/1977#issuecomment-402213517
Should this be closed as a duplicate of that?

**Comment by uranusjr** [comment]
I would say `sync` should stop looking for Pipfile.

**Comment by ulrikjohansson** [comment]
I've tinkered a little with just copying the find() code from the patched pipfile module and using that for finding the lock-file in `pipenv/project.py`, and that seems to solve the problem of not finding the lockfile, but since `do_sync()` runs `ensure_project()`, an empty Pipfile is always created when running the sync command.
Getting rid of all the uses of Pipfile in all the functions called by `do_sync()` quickly grew to a much larger project, so I'm not sure that should be included in a fix for this issue. Perhaps that should go with solving #41 that you mentioned above.
Is it reasonable to post a PR just fixing the lockfile find problem? In that case I'll try and get one ready within the next day or two.

**Comment by dudil** [comment]
Hi - Is that still open? I'll be happy to assist with that if help is still required

**Comment by uranusjr** [comment]
This is open.

**Comment by ulrikjohansson** [comment]
It looks like this issue is at least partly solved by https://github.com/pypa/pipenv/pull/3386,
I no longer get an error about a missing Pipfile or lockfile when I do a `pipenv sync` with only the Pipfile.lock present.

There is still weirdness remaining tough, like the `pipenv sync` creates a Pipfile just like when running `pipenv install --ignore-pipfile` , that I described in a comment further up: https://github.com/pypa/pipenv/issues/2834#issuecomment-419699150
So these two commands seem to behave the same way now.

The subject of this issue only mentions the sync failing, which is no longer the case. Should this issue be closed?

**Comment by frostming** [comment]
Sure thing, thank you for following up
